File: DEVOPSjavaPhyton/CajeroJava/src/main/java/com/mycompany/cajero/Cajero.java

package com.mycompany.cajero;

import java.awt.HeadlessException;
import javax.swing.JOptionPane;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

/**
 *
 * @author vicher
 */
public class Cajero {

    
    
    public static void main(String[] args) {
        
       menu();
    }
    
    
    
    
    //valores
    static double monto=1000;
    static double cantidad=0;
    static String historial="";
    
    
    static String fecha(){
         DateTimeFormatter fecha = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm");
        return fecha.format(LocalDateTime.now());
    }
    
    //contraseña
    static void contrasenna(){        
        int contador =1;
        while (contador <= 3) {            
            try {
                int x= Integer.parseInt(JOptionPane.showInputDialog("Ingresa tu pin"));
                if(x == 1235){
                    //JOptionPane.showMessageDialog(null, "Pin correcto");
                    break;
                }else if( contador == 3){
                        System.exit(0);
                }else{
                    JOptionPane.showMessageDialog(null, "Pin incorrecto, te restan :"+ (3-contador) + " intentos mäs");
                    contador ++;                    
                }

            } catch (HeadlessException | NumberFormatException w) {
                JOptionPane.showMessageDialog(null, "Pin incorrecto, te restan :"+ (3-contador) + " intentos mäs");
                    contador ++;   
                System.err.println(" "+w);
                

            } 
        }
    }
    
    //menu
    static void menu(){
        contrasenna();
            int opcion=0;
            while (true) { 
                try {
                   opcion = Integer.parseInt(JOptionPane.showInputDialog(" Bienvenido al cajero automatico\n" +
                    "\n" + "******Menú******\n" +
                    " 1-  Depositar\n" +
                    " 2- Retirar\n" +
                    " 3- Consultar saldo\n" +
                    " 4- Historial \n"+
                    " 5- Salir ")); 
                } catch (Exception e) {
                    System.err.print(e);
                }
                
                switch (opcion) {
                    case 1:
                        Depositar();
                        break;
                    case 2:
                        Retirar();
                        break;
                    case 3:
                        Ver();
                        break;
                    case 4:
                        Historial();
                        break;
                    case 5:
                        System.exit(0);
                        break;
                    default:JOptionPane.showMessageDialog(null, "Opcion invalida vuelve a intentar");
                        //throw new AssertionError();
                }
            }
       }
    
    
    //Metodos
    static void Depositar(){
        try {
            cantidad=Double.parseDouble(JOptionPane.showInputDialog("Introduce el monto a ingresar"));
            JOptionPane.showMessageDialog(null, "Operacion exitosa");
            //JOptionPane.showMessageDialog(null,"Usted deposito: $"+ cantidad );
            historial+= " "+fecha()+ " ingreso $"+ cantidad +" M/N\n";
            monto+=cantidad;
           // JOptionPane.showMessageDialog(null,"y su saldo es de: "+ monto);
        } catch (Exception e) {
            JOptionPane.showMessageDialog(null,"Valores invalidos");
            System.err.println("Texto invalido "+e);
        }  
    }
    
    static void Retirar(){
        try {
            cantidad=Double.parseDouble(JOptionPane.showInputDialog("Introduce el monto a retirar"));
            //JOptionPane.showMessageDialog(null,"Usted retiro: $"+ cantidad +"\n");
            if(monto >= cantidad){
            JOptionPane.showMessageDialog(null, "Operacion exitosa");
                historial+= " "+fecha()+ " retiro $"+ cantidad +" M/N\n";
                monto-=cantidad;
                //JOptionPane.showMessageDialog(null,"y su saldo es de: "+ monto);
            }else{
                JOptionPane.showMessageDialog(null, "Fondos insuficientes");
            }
        } catch (Exception a) {
            JOptionPane.showMessageDialog(null,"Valores invalidos");
            System.err.println("Texto invalido "+a);
        }  
    }
     
     static void Ver(){
        JOptionPane.showMessageDialog(null, "Tu saldo es de: $"+monto);
    }
    static void Historial(){
        if (historial != ""){
            JOptionPane.showMessageDialog(null, "El historial es :\n"+ historial);
        }else{
            JOptionPane.showMessageDialog(null, "No hay registros");
        }
        
        
    }
    
}

File: Cajero.py
# -*- coding: utf -8 -*-
import os

from datetime import datetime


class Cajero:

    def __init__(self):
        self.estado = True
        self.monto = 1000
        self.menu()
        self.historial=""
        self.imprimir()

    def contraseña(self):
        contador = 1
        while contador <= 3:
            try:
                x = int(input("Ingrese su pin o contraseña:"))
                if x == 1235:
                   # print("Pin Correcto")
                    break
                else:
                    print(f"Pin incorrecto, restan {3 - contador} intentos más")

                    if contador == 3:
                        self.estado = False
                        break
                    contador += 1

            except ValueError:
                print(f"Pin incorrecto, restan {3 - contador} intentos más")
                contador += 1
                print("Ingrese solo numeros")

    def menu(self):
        self.contraseña()
        print("Bienvenido cliente Victor Manuel")
        opcion = 0
        while opcion != "5":
            os.system("cls")
            print(""" Bienvenido al cajero automatico Victor Manuel
            ******Menú******
            1-  Depositar
            2- Retirar
            3- Consultar saldo
            4- Historial
            5- Salir """)
            opcion = input("Introdusca el número de operacion a realizar: ")
            if self.estado:
                if opcion == "1":
                    self.depositar()
                elif opcion == "2":
                    self.retiro()
                elif opcion == "3":
                    self.ver()
                elif opcion == "4":
                    self.imprimir()
                elif opcion == "5":
                    print("Programa finalizado")
                else:
                    print("NO existe esa opción, vuelva a intentar")

    def depositar(self):
        deposito = float(input("Ingrese la cantidad a depositar:"))
        self.registroHistorial()
        print("Usted a depositado $", deposito)
        print(f"Su  saldo es de: $ {self.monto + deposito}")
        self.monto += deposito

    def retiro(self):
        retiro = float(input("Ingrese la cantidad a retirar : "))
        if self.monto >= retiro:
            # self.historial()
            print(f"Usted a retirado: ${retiro} , su nuevo saldo es de ${self.monto - retiro}")
            self.monto -= retiro
        else:
            print("Imposible realizar el retiro, su monto es menor")

    def ver(self):
        print("Su saldo es de: $", self.monto)

    def registroHistorial(self):
        date = datetime.now()
        self.historial = date.strftime("%d %B, %Y %H:%M:%S"), "$",self.monto
        #print(self.historial)

    def imprimir(self):
        print(type(self.historial))
        if(self.historial != ""):
            print(type( self.historial))
        else:
            print("No existe registro")


Cajero()
